cypress/e2e/layers.cy.ts

```diff
@@ -160,114 +147,138 @@ describe("layers", () => {
+                },
+              ],
+            });
+          });
         });
 
-        it("comments", () => {
-          let bgId = createBackground();
+        describe("comments", () => {
+          let bgId: string;
           let comment = "42";
 
-          when.click("layer-list-item:background:" + bgId);
-          when.setValue("layer-comment.input", comment);
-
-          when.click("layer-editor.layer-id");
-
-          should.equalStyleStore(
-            (a: any) => a.layers,
-            [
-              {
-                id: "background:" + bgId,
-                type: "background",
-                metadata: {
-                  "maputnik:comment": comment,
+          beforeEach(() => {
+            bgId = createBackground();
+            when.click("layer-list-item:background:" + bgId);
+            when.setValue("layer-comment.input", comment);
+            when.click("layer-editor.layer-id");
+          });
+
+          it("should update style in local storage", () => {
+            then(get.styleFromLocalStorage()).shouldDeepNestedInclude({
+              layers: [
+                {
+                  id: "background:" + bgId,
+                  type: "background",
+                  metadata: {
+                    "maputnik:comment": comment,
+                  },
                 },
-              },
-            ]
-          );
-
-          // Unset it again.
-          when.typeKeys("{backspace}{backspace}", "layer-comment.input");
-          when.click("min-zoom.input-text");
-
-          should.equalStyleStore((a: any) => a.layers, [{
-            "id": 'background:' + bgId,
-            "type": 'background'
-          }]);
+              ],
+            });
+          });
+
+          describe("when unsetting", () => {
+            beforeEach(() => {
+              when.clear("layer-comment.input");
+              when.click("min-zoom.input-text");
+            });
+
+            it("should update style in local storage", () => {
+              then(get.styleFromLocalStorage()).shouldDeepNestedInclude({
+                layers: [
+                  {
+                    id: "background:" + bgId,
+                    type: "background",
+                  },
+                ],
+              });
+            });
+          });
         });
 
-        it("color", () => {
-          let bgId = createBackground();
-
-          when.click("layer-list-item:background:" + bgId);
-
-          when.click("spec-field:background-color");
-
-          should.equalStyleStore(
-            (a: any) => a.layers,
-            [
-              {
-                id: "background:" + bgId,
-                type: "background",
-              },
-            ]
-          );
+        describe("color", () => {
+          let bgId: string;
+          beforeEach(() => {
+            bgId = createBackground();
+            when.click("layer-list-item:background:" + bgId);
+            when.click("spec-field:background-color", 0);
```

Comment:
Isn't 0 the default? Is this for readability?

Reply:
Nope. Change the default to undefined, otherwise, all the checks of "non existence" will fail.
In this specific case there are two identical data attributes for different elements. Once this is fixed we can remove the 0 here 
![image](https://github.com/maputnik/editor/assets/60476837/827ff3e5-eeef-40c5-8882-e31d01056270)

Reply:
Is it possible to donit as part of this PR?

Reply:
Well, I am not very familiar with the code, which one of these does the test intend to use?

Reply:
I've added a commit to resolve this. The `FieldFunction` is a wrapper whereas the `InpucSpec` is a more down the hierarchy, which is what the test needs.
I'm not very familiar with the code either...